franklin/builder/serializers.py

```diff
@@ -27,6 +39,43 @@ class Meta:
         }
 
 
+class EnvironmentDetailSerializer(serializers.ModelSerializer):
+
+    def to_representation(self, data):
+        current_deploy = data.current_deploy
+        build_result = {}
+        if not current_deploy:
+            if self.context and self.context.get('user', None):
+                user = self.context['user']
+                branch, git_hash = data.site.get_newest_commit(user)
+                build_result = {
+                        'default_branch': branch,
+                        'git_hash': git_hash
+                }
+        elif hasattr(current_deploy, 'branchbuild'):
+            build = BranchBuild.objects.get(created=current_deploy.created)
+            serializer = BranchBuildSerializer(build)
+            build_result = serializer.data
+        elif hasattr(current_deploy, 'tagbuild'):
+            build = TagBuild.objects.get(created=current_deploy.created)
+            serializer = TagBuildSerializer(build)
+            build_result = serializer.data
+
+        env_serializer = EnvironmentSerializer(data)
+        result = env_serializer.data
+        result['current_deploy'] = build_result
+        return result
```

Comment:
this `to_representation` in interesting. We use different serializers based on which model type the `current_deploy` is. For an environment that does not have an current deployment, we call github and return the default branch and it's current tip commit instead. All of this is done from the serializer. Feels a little bloated. Will look at this with Chad and Solomon further.